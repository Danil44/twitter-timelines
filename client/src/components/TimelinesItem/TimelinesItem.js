import React from 'react';
import {
  Card,
  CardHeader,
  Avatar,
  Typography,
  CardContent,
} from '@material-ui/core';
import FavoriteIcon from '@material-ui/icons/Favorite';
import PropTypes from 'prop-types';
import ShareIcon from '@material-ui/icons/Share';
import useStyles from './TimelinesItemStyles';
import TimelinesQuotedTweet from '../TimelinesQuotedTweet/TimelinesQuotedTweet';

const TimelinesItem = ({
  text,
  name,
  createdAt,
  screenName,
  profileImage,
  retweets,
  likes,
  quotedTweet,
}) => {
  const classes = useStyles();

  return (
    <Card className={classes.card}>
      <CardHeader
        avatar={<Avatar aria-label={name} src={profileImage} />}
        title={name}
        titleTypographyProps={{
          variant: 'h6',
          display: 'inline',
          className: classes.name,
        }}
        subheaderTypographyProps={{ display: 'inline' }}
        subheader={`@${screenName} . ${createdAt}`}
      />
      <CardContent>
        <Typography variant="body1">{text}</Typography>

        {quotedTweet && <TimelinesQuotedTweet {...quotedTweet} />}
      </CardContent>
      <div className={classes.iconContainer}>
        <FavoriteIcon
          stroke="grey"
          strokeWidth={2}
          className={classes.likeIcon}
        />
        <Typography variant="body2" className={classes.count}>
          {likes}
        </Typography>
        <ShareIcon className={classes.icon} />
        <Typography variant="body2" className={classes.count}>
          {retweets}
        </Typography>
      </div>
    </Card>
  );
};

TimelinesItem.defaultProps = {
  quotedTweet: null,
};

TimelinesItem.propTypes = {
  text: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  screenName: PropTypes.string.isRequired,
  profileImage: PropTypes.string.isRequired,
  likes: PropTypes.number.isRequired,
  retweets: PropTypes.number.isRequired,
  createdAt: PropTypes.string.isRequired,
  quotedTweet: PropTypes.shape({}),
};

export default TimelinesItem;
