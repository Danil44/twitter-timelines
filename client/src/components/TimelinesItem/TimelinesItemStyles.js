import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  card: {
    maxWidth: '100%',
  },

  name: {
    marginRight: 5,
  },

  iconContainer: {
    display: 'flex',
    alignItems: 'center',
    padding: '0 0 10px 10px',
  },

  likeIcon: {
    margin: theme.spacing(1),
    color: 'transparent',
  },

  icon: {
    margin: theme.spacing(1),
    color: 'grey',
  },

  count: {
    marginRight: 15,
  },
}));

export default useStyles;
