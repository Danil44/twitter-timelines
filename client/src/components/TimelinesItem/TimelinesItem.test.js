import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TimelinesItem from './TimelinesItem';

describe('<TimelinesItem/> component', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      text: 'text',
      name: 'name',
      createdAt: 'date',
      screenName: 'name',
      profileImage: 'image url',
      retweets: 0,
      likes: 0,
      quotedTweet: {
        user: {
          name: 'name',
          screen_name: 'screen name',
          profile_image_url: 'profile image url',
        },
        created_at: 'date',
        text: 'text',
      },
    };

    wrapper = shallow(<TimelinesItem {...props} />);
  });

  it('should renders without crashing given the required props', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
