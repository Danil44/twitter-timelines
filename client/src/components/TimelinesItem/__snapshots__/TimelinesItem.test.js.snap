// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelinesItem/> component should renders without crashing given the required props 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-card-1"
>
  <WithStyles(ForwardRef(CardHeader))
    avatar={
      <ForwardRef(WithStyles)
        aria-label="name"
        src="image url"
      />
    }
    subheader="@name . date"
    subheaderTypographyProps={
      Object {
        "display": "inline",
      }
    }
    title="name"
    titleTypographyProps={
      Object {
        "className": "makeStyles-name-2",
        "display": "inline",
        "variant": "h6",
      }
    }
  />
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      variant="body1"
    >
      text
    </WithStyles(ForwardRef(Typography))>
    <TimelinesQuotedTweet
      created_at="date"
      text="text"
      user={
        Object {
          "name": "name",
          "profile_image_url": "profile image url",
          "screen_name": "screen name",
        }
      }
    />
  </WithStyles(ForwardRef(CardContent))>
  <div
    className="makeStyles-iconContainer-3"
  >
    <FavoriteIcon
      className="makeStyles-likeIcon-4"
      stroke="grey"
      strokeWidth={2}
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-count-6"
      variant="body2"
    >
      0
    </WithStyles(ForwardRef(Typography))>
    <ShareIcon
      className="makeStyles-icon-5"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-count-6"
      variant="body2"
    >
      0
    </WithStyles(ForwardRef(Typography))>
  </div>
</WithStyles(ForwardRef(Card))>
`;
